apps/namadillo/src/hooks/useTransaction.tsx

```diff
@@ -100,58 +98,62 @@ export const useTransaction = <T,>({
   const execute = async (
     txAdditionalParams: Partial<BuildTxAtomParams<T>> = {}
   ): Promise<TransactionPair<T> | void> => {
-    try {
-      invariant(gasConfig.data, "Gas config not loaded");
-      invariant(
-        account?.address,
-        "Extension not connected or no account is selected"
-      );
-
-      const tx = await buildTx({
-        params,
-        gasConfig: gasConfig.data,
-        account,
-        ...txAdditionalParams,
-      });
-
-      if (!tx) throw "Error: invalid TX created by buildTx";
-      if (onSigned) {
-        onSigned(tx);
-      }
+    invariant(selectedGasConfig, "Gas config not loaded");
+    invariant(
+      account?.address,
+      "Extension not connected or no account is selected"
+    );
 
-      await broadcast(tx);
+    const tx = await buildTx({
+      params,
+      gasConfig: selectedGasConfig,
+      account,
+      ...txAdditionalParams,
+    });
 
-      if (parsePendingTxNotification) {
-        dispatchPendingTxNotification(tx, parsePendingTxNotification(tx));
-      }
+    if (!tx) throw "Error: invalid TX created by buildTx";
+    if (onSigned) {
+      onSigned(tx);
+    }
 
-      if (onSuccess) {
-        onSuccess(tx);
-      }
+    if (parsePendingTxNotification) {
+      dispatchPendingTxNotification(tx, parsePendingTxNotification(tx));
+    }
 
-      return tx;
-    } catch (err) {
-      if (parseErrorTxNotification) {
-        dispatchErrorNotification(err, parseErrorTxNotification());
-      }
+    broadcastTxWithEvents(
+      tx.encodedTxData,
+      tx.signedTxs,
+      tx.encodedTxData.meta?.props,
+      eventType
+    )
+      .then(() => {
+        onSuccess?.(tx);
+      })
+      .catch((err) => {
+        if (parseErrorTxNotification) {
+          dispatchErrorNotification(err, parseErrorTxNotification());
+        }
+
+        if (onError) {
+          onError(err);
+        } else {
+          throw err;
+        }
+      });
 
-      if (onError) {
-        onError(err);
-      } else {
-        throw err;
-      }
-    }
+    onBroadcasted?.();
```

Comment:
We don't have an `await` on the `broadcastTxWithEvents`, so I'm thinking if we should call it like `onStart` or something similar? 

Or, maybe, the `onSuccess` should be renamed to `onBroadcasted`